swarm/api/api.go

```diff
@@ -654,72 +657,79 @@ func (self *Api) BuildDirectoryTree(mhash string, nameresolver bool) (addr stora
 	return addr, manifestEntryMap, nil
 }
 
-// Look up mutable resource updates at specific periods and versions
-func (self *Api) ResourceLookup(ctx context.Context, addr storage.Address, period uint32, version uint32, maxLookup *mru.LookupParams) (string, []byte, error) {
+// ResourceLookup - Look up mutable resource updates at specific periods and versions
+func (a *API) ResourceLookup(ctx context.Context, addr storage.Address, period uint32, version uint32, maxLookup *mru.LookupParams) (string, []byte, error) {
 	var err error
-	rsrc, err := self.resource.Load(addr)
+	rsrc, err := a.resource.Load(addr)
 	if err != nil {
 		return "", nil, err
 	}
 	if version != 0 {
 		if period == 0 {
 			return "", nil, mru.NewError(mru.ErrInvalidValue, "Period can't be 0")
 		}
-		_, err = self.resource.LookupVersion(ctx, rsrc.NameHash(), period, version, true, maxLookup)
+		_, err = a.resource.LookupVersion(ctx, rsrc.NameHash(), period, version, true, maxLookup)
 	} else if period != 0 {
-		_, err = self.resource.LookupHistorical(ctx, rsrc.NameHash(), period, true, maxLookup)
+		_, err = a.resource.LookupHistorical(ctx, rsrc.NameHash(), period, true, maxLookup)
 	} else {
-		_, err = self.resource.LookupLatest(ctx, rsrc.NameHash(), true, maxLookup)
+		_, err = a.resource.LookupLatest(ctx, rsrc.NameHash(), true, maxLookup)
 	}
 	if err != nil {
 		return "", nil, err
 	}
 	var data []byte
-	_, data, err = self.resource.GetContent(rsrc.NameHash().Hex())
+	_, data, err = a.resource.GetContent(rsrc.NameHash().Hex())
 	if err != nil {
 		return "", nil, err
 	}
 	return rsrc.Name(), data, nil
 }
 
-func (self *Api) ResourceCreate(ctx context.Context, name string, frequency uint64) (storage.Address, error) {
-	key, _, err := self.resource.New(ctx, name, frequency)
+// ResourceCreate - create's Resource and returns it's key
+func (a *API) ResourceCreate(ctx context.Context, name string, frequency uint64) (storage.Address, error) {
+	key, _, err := a.resource.New(ctx, name, frequency)
 	if err != nil {
 		return nil, err
 	}
 	return key, nil
 }
 
-func (self *Api) ResourceUpdateMultihash(ctx context.Context, name string, data []byte) (storage.Address, uint32, uint32, error) {
-	return self.resourceUpdate(ctx, name, data, true)
+// ResourceUpdateMultihash - updates Multihash resource
```

Comment:
ResourceUpdateMultihash updates Multihash-based mutable resource update

Reply:
ResourceUpdateMultihash updates a Mutable Resource and marks the update's content to be of multihash type, which will be recognized upon retrieval.
It will fail if the data is not a valid multihash.